pkg/contextstore/store.go

```diff
@@ -0,0 +1,290 @@
+package contextstore
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const configFileName = "config.json"
+
+// Store provides a context store for easily remembering endpoints configuration
+type Store interface {
+	GetCurrentContext() string
+	SetCurrentContext(name string) error
+	ListContexts() (map[string]ContextMetadata, error)
+	CreateOrUpdateContext(name string, meta ContextMetadata) error
+	GetContextMetadata(name string) (ContextMetadata, error)
+	ResetContextTLSMaterial(name string, data *ContextTLSData) error
+	ResetContextEndpointTLSMaterial(contextName string, endpointName string, data *EndpointTLSData) error
+	ListContextTLSFiles(name string) (map[string]EndpointFiles, error)
+	GetContextTLSData(contextName, endpointName, fileName string) ([]byte, error)
+	Export(name string) io.ReadCloser
```

Comment:
I think those 3 methods `Export`/`Import`/`Remove` shouldn't be part of the interface. They can be only exported helper functions:
```go
func Export(name string, store Store) io.ReadCloser{
    ...
}
...
```